src/typing.py

```diff
@@ -1859,6 +1860,21 @@ def __new__(cls, *args, **kwds):
                             "create a subclass instead")
         return _generic_new(_G_base, cls, *args, **kwds)
 
+if hasattr(collections_abc, 'AsyncGenerator'):
+    _AG_base = collections_abc.AsyncGenerator
+
+    class AsyncGenerator(AsyncIterator[T_co], Generic[T_co, T_contra],
+                         extra=_AG_base):
+        __slots__ = ()
+
+        def __new__(cls, *args, **kwds):
```

Comment:
I don't think I understand your point. The only way to instantiate a subclass of an ABC is to implement all abstract methods. Moreover, there is no way a subclass could "use" this ``__new__``, it only exists everywhere in ``typing`` to prohibit direct instantiation of generic version of concrete collections and similar things.

``__new__`` for ``Generator`` is necessary, since it could inherit from ``types.GeneratorType``.